crates/miden-lib/asm/kernels/transaction/lib/epilogue.masm

```diff
@@ -306,95 +335,101 @@ end
 #!   delta commitment.
 #! - FEE_ASSET is the fungible asset used as the transaction fee.
 #! - tx_expiration_block_num is the transaction expiration block number.
-export.finalize_transaction.4
+#!
+#! Locals:
+#! - 0..4: OUTPUT_NOTES_COMMITMENT
+#! - 4..8: FEE_ASSET
+export.finalize_transaction.8
     # make sure that the context was switched back to the native account
     exec.memory::assert_native_account
 
-    # execute the account authentication procedure
+    # ------ Execute the account's authentication procedure ------
+
     exec.execute_auth_procedure
     # => []
 
+    # ------ Update account code ------
+
     # update the account code commitment in case its code has been changed
     exec.update_code_commitment
     # => []
 
-    # compute the fee the tx needs to pay
-    # ignored for now
-    exec.compute_fee
-    # => [fee_amount]
+    # ------ Build output vault ------
 
-    exec.build_native_fee_asset
-    # => [FEE_ASSET]
+    exec.build_output_vault
+    # => []
 
-    # remove the fee from the native account's vault
-    exec.account::remove_asset_from_vault
-    # => [FEE_ASSET]
+    # ------ Compute output notes commitment ------
+
+    exec.note::compute_output_notes_commitment
+    # => [OUTPUT_NOTES_COMMITMENT]
 
-    # store the fee asset for later
+    # copy output note data to the advice map
+    exec.copy_output_notes_to_advice_map
+    # => [OUTPUT_NOTES_COMMITMENT]
+
+    # store commitment in local
     loc_storew.0 dropw
     # => []
 
-    # get the initial account commitment
-    exec.memory::get_init_account_commitment
-    # => [INIT_ACCOUNT_COMMITMENT]
+    # ------ Compute fees ------
+
+    exec.compute_and_remove_fee
+    # => [FEE_ASSET]
+
+    # store fee asset in local
+    loc_storew.4 dropw
+    # => []
```

Comment:
I would maybe add a short comment somewhere here to make it super clear that we need to compute and remove the fee before we compute final account commitment (it is clear already - but stressing this probably doesn't hurt).